src/main/java/uk/nhs/hee/tis/trainee/reference/api/GradeResource.java

```diff
@@ -73,6 +76,12 @@ public GradeResource(GradeService service, GradeMapper mapper) {
    */
   @PostMapping("/grade")
   public ResponseEntity<GradeDto> createGrade(@RequestBody GradeDto gradeDto) {
+    if (!validator.isValid(gradeDto)) {
+      // TODO: refactor service to take DTO instead of entity and move business logic there?
+      service.deleteByTisId(gradeDto.getTisId());
```

Comment:
I'm sure there's a good reason for this, but isn't it a little surprising to delete an existing entry from the database if an invalid creation is posted? I might have thought we'd leave the existing record in the database as-is

Reply:
A full load from DMS is treated as a POST, so this covers that scenario where we need to remove an invalid grade from a previous sync.

Reply:
I can see how we could have invalid grades in the database, since they won't have the newly defined fields, but I imagined the new sync will only have the valid records. Wouldn't the old invalid ones simply be overwritten?

Reply:
Lets say we have Grade 123

Initially received by `tis-trainee-reference` as a valid grade (`CURRENT`, `placementGrade = true` and `trainingGrade = true`)
```
{
  "id": 123,
  "label": "ST1",
  "placementGrade": true,
  "trainingGrade": true,
  "status": "CURRENT"
}
```

Stored in `tis-trainee-reference` db as
```
{
  "id": 123,
  "label": "ST1"
}
```

A change is then made in TIS which sets `placementGrade = false`, this is received by `tis-trainee-reference` and flagged "invalid"
```
{
  "id": 123,
  "label": "ST1",
  "placementGrade": false,
  "trainingGrade": true,
  "status": "CURRENT"
}
```

We don't want to overwrite the existing record in the `tis-trainee-reference` db because the new record isn't valid.
We also don't want to leave the old record, because it's outdated.
So we delete it instead.

Reply:
Ah ok, that makes sense. I had not imagined that grades could move from a valid to an invalid state in TIS.